src/ucs/sys/iovec.c

```diff
@@ -40,10 +42,13 @@ size_t ucs_iov_copy(const struct iovec *iov, size_t iov_cnt,
         len     -= iov_offset;
 
         len = ucs_min(len, max_copy);
+
         if (dir == UCS_IOV_COPY_FROM_BUF) {
-            memcpy(iov_buf, UCS_PTR_BYTE_OFFSET(buf, copied), len);
+            ucs_memcpy_relaxed(iov_buf, UCS_PTR_BYTE_OFFSET(buf, copied), len,
```

Comment:
here we might need to hint at least with the original/non-updated length `iov[i].iov_len`?

Reply:
@tvegas1 total_len parameter to ucs_memcpy_relaxed() should be  iov[i].iov_len ?

Reply:
re-reading, actually using `len` here was right